packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte

```diff
@@ -0,0 +1,302 @@
+    const providerContainer = await Promise.all(
+      provider.containerConnections.map(async container => {
+        const containerConfigurations = await Promise.all(
+          configurationKeys.map(async configurationKey => {
+            return {
+              ...configurationKey,
+              value: await window.getConfigurationValue(
+                configurationKey.id,
+                container as unknown as ContainerProviderConnection,
+              ),
+              container: container.name,
+              providerId: provider.internalId,
+            };
+          }),
+        );
+        return containerConfigurations;
+      }),
+    );
+    return providerContainer.flat();
+  }),
+).then(value => (tmpProviderContainerConfiguration = value.flat()));
+
+$: providerContainerConfiguration = tmpProviderContainerConfiguration
+  .filter(configurationKey => configurationKey.value !== undefined)
+  .reduce((map, value) => {
+    const innerProviderContainerConfigurations = map.get(value.providerId) || [];
+    innerProviderContainerConfigurations.push(value);
+    map.set(value.providerId, innerProviderContainerConfigurations);
+    return map;
+  }, new Map<string, IProviderContainerConfigurationPropertyRecordedSchema[]>());
+
+async function startContainerProvider(
+  provider: ProviderInfo,
+  containerConnectionInfo: ProviderContainerConnectionInfo,
+): Promise<void> {
+  if (containerConnectionInfo.status === 'stopped') {
+    setContainerStatusIsChanging(provider, containerConnectionInfo);
+    await window.startProviderConnectionLifecycle(provider.internalId, containerConnectionInfo);
+  }
+}
+
+async function stopContainerProvider(
+  provider: ProviderInfo,
+  containerConnectionInfo: ProviderContainerConnectionInfo,
+): Promise<void> {
+  if (containerConnectionInfo.status === 'started') {
+    setContainerStatusIsChanging(provider, containerConnectionInfo);
+    await window.stopProviderConnectionLifecycle(provider.internalId, containerConnectionInfo);
+  }
+}
+
+async function restartContainerProvider(
+  provider: ProviderInfo,
+  containerConnectionInfo: ProviderContainerConnectionInfo,
+): Promise<void> {
+  await stopContainerProvider(provider, containerConnectionInfo);
+  await startContainerProvider(provider, containerConnectionInfo);
+}
+
+async function deleteContainerProvider(
+  provider: ProviderInfo,
+  containerConnectionInfo: ProviderContainerConnectionInfo,
+): Promise<void> {
+  if (containerConnectionInfo.status === 'stopped' || containerConnectionInfo.status === 'unknown') {
+    setContainerStatusIsChanging(provider, containerConnectionInfo);
+    await window.deleteProviderConnectionLifecycle(provider.internalId, containerConnectionInfo);
+  }
+}
+
+function setContainerStatusIsChanging(
+  provider: ProviderInfo,
+  containerConnectionInfo: ProviderContainerConnectionInfo,
+): void {
+  containerConnectionStatus.set(`${provider.name}-${containerConnectionInfo.name}`, {
```

Comment:
probably need an helper that produces the key to be sure it's consistent over the class (one could use `-`, another `.`, etc)